reconcile/dynatrace_token_provider/dependencies.py

```diff
@@ -1,3 +1,5 @@
+from __future__ import annotations
```

Comment:
where is this used?

Reply:
In the creator method

```
def create(cls, secret_reader: SecretReaderBase) -> Dependencies
```

Else Id need to wrap it the type with ticks as forward declaration. But I prefer it without.

Reply:
can use `Self`, `def create(cls, secret_reader: SecretReaderBase) -> Self`

Reply:
`Self` doesnt work for me - ruff throwing error. Whats the problem with my current approach?

Reply:
Ah forgot the `Self` import. But still, why not my current approach?

Reply:
because that is the new feature in python 3.11 to solve this issue https://peps.python.org/pep-0673/, `__future__. annotations ` is still uncertain (TBD).

Reply:
I see. Adjusted :+1: